x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -0,0 +1,295 @@
+      } as const;
+    });
+
+export const GcpCredentialsForm = ({
+  input,
+  newPolicy,
+  updatePolicy,
+  packageInfo,
+  setIsValid,
+  onChange,
+}: Props) => {
+  const fields = getInputVarsFields(input, gcpField.fields);
+
+  useEffect(() => {
+    const isInvalid = packageInfo.version < MIN_VERSION_GCP_CIS;
+
+    setIsValid(!isInvalid);
+
+    onChange({
+      isValid: !isInvalid,
+      updatedPolicy: newPolicy,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [input, packageInfo]);
+
+  if (packageInfo.version < MIN_VERSION_GCP_CIS) {
+    return (
+      <>
+        <EuiSpacer size="l" />
+        <EuiCallOut color="warning">
+          <FormattedMessage
+            id="xpack.csp.gcpIntegration.gcpNotSupportedMessage"
+            defaultMessage="CIS GCP is not supported on the current Integration version, please upgrade your integration to the latest version to use CIS GCP"
+          />
+        </EuiCallOut>
+      </>
+    );
+  }
+  return (
+    <>
+      <GCPSetupInfoContent />
+      <EuiSpacer size="l" />
+      <GcpSetupAccessSelector
+        onChange={(optionId) => updatePolicy(getPosturePolicy(newPolicy, input.type))}
+      />
+      <EuiSpacer size="l" />
+      <GcpInputVarFields
+        fields={fields}
+        onChange={(key, value) =>
+          updatePolicy(getPosturePolicy(newPolicy, input.type, { [key]: { value } }))
+        }
+      />
+      <EuiSpacer size="s" />
+      {DocsLink}
+      <EuiSpacer />
+    </>
+  );
+};
+
+const GcpSetupAccessSelector = ({ onChange }: { onChange(type: GcpCredentialsType): void }) => (
+  <RadioGroup
+    size="s"
+    options={getSetupFormatOptions()}
+    idSelected={'manual'}
+    onChange={(id: GcpCredentialsType) => onChange(id)}
+  />
+);
+
+const GcpInputVarFields = ({
+  fields,
+  onChange,
+}: {
+  fields: Array<GcpFields[keyof GcpFields] & { value: string; id: string }>;
+  onChange: (key: string, value: string) => void;
+}) => {
+  const [credentialOption, setCredentialOption] = useState('Credentials File');
+  const targetFieldName = (id: string) => {
+    return fields.find((element) => element.id === id);
+  };
+  return (
+    <div>
+      <EuiForm component="form">
+        <EuiFormRow fullWidth label={gcpField.fields.project_id.label}>
+          <EuiFieldText
+            id={targetFieldName('project_id')!.id}
+            fullWidth
+            value={targetFieldName('project_id')!.value || ''}
+            onChange={(event) => onChange(targetFieldName('project_id')!.id, event.target.value)}
+          />
+        </EuiFormRow>
+        <EuiFormRow fullWidth label={'Credentials'}>
+          <EuiSelect
+            fullWidth
+            options={credentialOptionsList}
+            value={credentialOption}
+            onChange={(optionElem) => {
+              setCredentialOption(optionElem.target.value);
+            }}
+          />
+        </EuiFormRow>
+        {credentialOption === 'Credentials File' && (
```

Comment:
this and the json option condition should use the `value` field's value after the change. might be useful to store in constants